src/riak_ql_inverse_distrib_fns.erl

```diff
@@ -0,0 +1,144 @@
+         fn_type_signature/2,
+         fn_param_check/2,
+         supported_functions/0]).
+
+-type invdist_function() :: 'PERCENTILE_CONT'
+                          | 'PERCENTILE_DISC'
+                          | 'MEDIAN'
+                          | 'MODE'.
+-export_type([invdist_function/0]).
+
+-include("riak_ql_ddl.hrl").
+
+supported_functions() ->
+    ['PERCENTILE_DISC', 'PERCENTILE_CONT', 'MEDIAN', 'MODE'].
+
+-spec fn_type_signature(invdist_function(), [riak_ql_ddl:external_field_type()]) ->
+                               riak_ql_ddl:external_field_type() |
+                               {error, term()}.
+fn_type_signature('PERCENTILE_DISC', [ColumnType, double])
+  when ColumnType == sint64;
+       ColumnType == double;
+       ColumnType == timestamp -> ColumnType;
+fn_type_signature('PERCENTILE_CONT', [_ColumnType, double])
+  when _ColumnType == sint64;
+       _ColumnType == double;
+       _ColumnType == timestamp -> double;
+fn_type_signature('MEDIAN', [ColumnType])
+  when ColumnType == sint64;
+       ColumnType == double;
+       ColumnType == timestamp  -> ColumnType;
+fn_type_signature('MODE', [ColumnType])
+  when ColumnType == sint64;
+       ColumnType == double;
+       ColumnType == timestamp  -> ColumnType;
+fn_type_signature(Fn, Args) ->
+    {error, {argument_type_mismatch, Fn, Args}}.
+
+-spec fn_arity(invdist_function()) -> non_neg_integer().
+fn_arity('PERCENTILE_CONT') -> 2;
+fn_arity('PERCENTILE_DISC') -> 2;
+fn_arity('MEDIAN') -> 1;
+fn_arity('MODE') -> 1;
+fn_arity(_) -> {error, invalid_function}.
+
+-spec fn_param_check(invdist_function(), [riak_ql_ddl:external_field_type()]) ->
+                            ok | {error, WhichParamInvalid::pos_integer()}.
+fn_param_check(PcntlFn, [Pc])
+  when (PcntlFn == 'PERCENTILE_CONT' orelse PcntlFn == 'PERCENTILE_DISC') andalso
+       (Pc >= 0.0 andalso Pc =< 1.0) ->
+    ok;
+fn_param_check(PcntlFn, [_Pc])
+  when (PcntlFn == 'PERCENTILE_CONT' orelse PcntlFn == 'PERCENTILE_DISC') ->
+    {error, 2};
+fn_param_check('MEDIAN', []) ->
+    ok;
+fn_param_check('MODE', []) ->
+    ok.
+
+
+%% functions defined
+%%
+%% Note that ValuesAtF expects row position to be 0-based.
+
+'PERCENTILE_DISC'([Pc], RowsTotal, ValuesAtF) ->
+    RN = (Pc * (RowsTotal - 1)),
+    [[Ret]] = ValuesAtF([{trunc(RN), 1}]),
+    Ret.
+
+'PERCENTILE_CONT'([Pc], RowsTotal, ValuesAtF) ->
+    RN = (Pc * (RowsTotal - 1)),
+    {LoRN, HiRN} = {trunc(RN), ceil(RN)},
+    case LoRN == HiRN of
+        true ->
+            [[Val]] = ValuesAtF([{LoRN, 1}]),
+            Val;
+        false ->
+            [[LoVal], [HiVal]] = ValuesAtF([{LoRN, 1}, {HiRN, 1}]),
+            (HiRN - RN) * LoVal + (RN - LoRN) * HiVal
+    end.
+
+'MEDIAN'([], RowsTotal, ValuesAtF) ->
+    'PERCENTILE_DISC'([0.5], RowsTotal, ValuesAtF).
+
+'MODE'([], RowsTotal, ValuesAtF) ->
+    [[Min]] = ValuesAtF([{0, 1}]),
+    largest_bin(Min, ValuesAtF, RowsTotal).
+
+%% this will be inefficient for ldb backends
```

Comment:
I think I've asked before but I'm tired: "ldb backends" ?

Please expand it, I won't be the only person confused about this comment in the future.